packages/libraries/cli/src/commands/whoami.ts

```diff
@@ -109,12 +120,31 @@ export default class WhoAmI extends Command<typeof WhoAmI> {
       });
 
       this.log(print());
-    } else if (result.tokenInfo.__typename === 'TokenNotFoundError') {
+
+      return this.successData({
+        data: {
+          tokenName: tokenInfo.token.name,
+          organization: organization.slug,
+          project: project.slug,
+          target: target.slug,
+          authorization: {
+            schema: {
+              publish: tokenInfo.canPublishSchema,
+              check: tokenInfo.canCheckSchema,
+            },
+          },
+        },
+      });
+    }
+
+    if (result.tokenInfo.__typename === 'TokenNotFoundError') {
       this.error(`Token not found. Reason: ${result.tokenInfo.message}`, {
         exit: 0,
         suggestions: [`How to create a token? https://docs.graphql-hive.com/features/tokens`],
       });
     }
+
+    casesExhausted(result.tokenInfo);
```

Comment:
You'll see this in a few places. It moves towards a stricter switch case pattern where new union cases should definitely raise a static error in our code.